addons/osfstorage/models.py

```diff
@@ -166,37 +176,37 @@ def check_in_or_out(self, user, checkout, save=False):
         from osf.models import NodeLog  # Avoid circular import
 
         if (
-                self.is_checked_out and self.checkout != user and permissions.ADMIN not in self.node.get_permissions(
+                self.is_checked_out and self.checkout != user and permissions.ADMIN not in self.target.get_permissions(
```

Comment:
Since `target` is a general type (and not always an `AbstractNode`), should check if `self.target` is an `AbstractNode` before calling `get_permissions()`.